src/containers/my-resources/add-resource-with-input/AddResourceWithInput.tsx

```diff
@@ -23,12 +22,12 @@ import AppButtonMenu from '~/components/app-button-menu/AppButtonMenu'
 import ResourcesToolBarDrawer from '~/containers/my-resources/resources-toolbar-drawer/ResourcesToolbarDrawer'
 
 import { styles } from '~/containers/my-resources/add-resource-with-input/AddResourceWithInput.styles'
-import { CategoryNameInterface, PositionEnum } from '~/types'
+import { type CategoryNameInterface, PositionEnum } from '~/types'
 import { InputFieldVariantEnum } from '~scss-components/input-field/InputField.constants'
 
 interface AddResourceWithInputProps {
   btnText?: string
-  fetchData: () => Promise<void>
+  fetchData: (() => Promise<void>) | (() => void)
```

Comment:
Does it work like that?

```suggestion
  fetchData: () => Promise<void> | void
```

Reply:
No, a TypeScript error occurs